int32 theta0
int32 theta1
int32 theta2
int32 theta3
